internal/cmd/mongodbflex/backup/restore/restore.go

```diff
@@ -0,0 +1,199 @@
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			model, err := parseInput(p, cmd)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			instanceLabel, err := mongodbUtils.GetInstanceName(ctx, apiClient, model.ProjectId, model.InstanceId)
+			if err != nil {
+				p.Debug(print.ErrorLevel, "get instance name: %v", err)
+				instanceLabel = model.ProjectId
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to restore MongoDB Flex instance %q?", instanceLabel)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// If backupInstanceId is not provided, the target is the same instance as the backup
+			if model.BackupInstanceId == "" {
+				model.BackupInstanceId = model.InstanceId
+			}
+
+			isRestoreOperation := getIsRestoreOperation(model.BackupId, model.Timestamp)
+
+			// If backupId is provided, restore the instance from the backup with the backupId
+			if isRestoreOperation {
+				req := buildRestoreRequest(ctx, model, apiClient)
+				_, err = req.Execute()
+				if err != nil {
+					return fmt.Errorf("restore MongoDB Flex instance: %w", err)
+				}
+
+				if !model.Async {
+					s := spinner.New(p)
+					s.Start("Restoring instance")
+					_, err = wait.RestoreInstanceWaitHandler(ctx, apiClient, model.ProjectId, model.InstanceId, model.BackupId).WaitWithContext(ctx)
+					if err != nil {
+						return fmt.Errorf("wait for MongoDB Flex instance restoration: %w", err)
+					}
+					s.Stop()
+				}
+
+				p.Outputf("Restored instance %q with backup %q\n", model.InstanceId, model.BackupId)
+				return nil
+			}
+
+			// Else, if timestamp is provided, clone the instance from a point-in-time snapshot
+			req := buildCloneRequest(ctx, model, apiClient)
+			_, err = req.Execute()
+			if err != nil {
+				return fmt.Errorf("clone MongoDB Flex instance: %w", err)
+			}
+
+			if !model.Async {
+				s := spinner.New(p)
+				s.Start("Cloning instance")
+				_, err = wait.CloneInstanceWaitHandler(ctx, apiClient, model.ProjectId, model.InstanceId).WaitWithContext(ctx)
+				if err != nil {
+					return fmt.Errorf("wait for MongoDB Flex instance cloning: %w", err)
+				}
+				s.Stop()
+			}
+
+			p.Outputf("Cloned instance %q from backup with timestamp %q\n", model.InstanceId, model.Timestamp)
+			return nil
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Var(flags.UUIDFlag(), instanceIdFlag, "Instance id")
+	cmd.Flags().Var(flags.UUIDFlag(), backupInstanceIdFlag, "Instance id of the target instance to restore the backup to")
+	cmd.Flags().String(backupIdFlag, "", "Backup id")
+	cmd.Flags().String(timestampFlag, "", "Timestamp of the snapshot to clone the instance from")
```

Comment:
we can mention the format and an example for the timestamp. Sth like this, 

, in a date-time with the RFC3339 layout format, e.g. 2024-01-01T00:00:00Z

similar to object-storage - credentials- create or postgresflex-instance-clone